### ISSUE TITLE
Error with undefined CSS variables in calc expression in shorthand property

### ISSUE THREAD

**Issue by Enteleform** [vivliostyle/vivliostyle.js#1028]
## Is your feature request related to a problem? Please describe.

I just ran into an issue in a PDF project after upgrading to the latest version of Astro. It turns out that they changed their asset naming algorithm, which was resulting in some missing CSS.

It seems that the reason Vivliostyle was being affected by this was that there were some CSS variables being used throughout the CSS that did make it into the build, but some of the variable definitions were missing.

This is the error that Vivliostyle displayed:

```
✖ TypeError: Cannot read properties of null (reading 'funcs')
 TypeError: Cannot read properties of null (reading 'funcs')
    at e.evalCall (file:///D:/DemoProject/node_modules/@vivliostyle/viewer/lib/js/vivliostyle-viewer.js:14:119825)  
    at n.evaluateCore (file:///D:/DemoProject/node_modules/@vivliostyle/viewer/lib/js/vivliostyle-viewer.js:14:128612)
    at e.evaluate (file:///D:/DemoProject/node_modules/@vivliostyle/viewer/lib/js/vivliostyle-viewer.js:14:121682)  
    at n.evaluateCore (file:///D:/DemoProject/node_modules/@vivliostyle/viewer/lib/js/vivliostyle-viewer.js:14:122751)
    at e.evaluate (file:///D:/DemoProject/node_modules/@vivliostyle/viewer/lib/js/vivliostyle-viewer.js:14:121682)  
    at n.visitFunc (file:///D:/DemoProject/node_modules/@vivliostyle/viewer/lib/js/vivliostyle-viewer.js:14:341068) 
    at n.visit (file:///D:/DemoProject/node_modules/@vivliostyle/viewer/lib/js/vivliostyle-viewer.js:14:138506)     
    at n.visitValues (file:///D:/DemoProject/node_modules/@vivliostyle/viewer/lib/js/vivliostyle-viewer.js:14:133119)
    at n.visitSpaceList (file:///D:/DemoProject/node_modules/@vivliostyle/viewer/lib/js/vivliostyle-viewer.js:14:133642)
    at n.visit (file:///D:/DemoProject/node_modules/@vivliostyle/viewer/lib/js/vivliostyle-viewer.js:14:137932)     
        ---- async ---
        StyleInstance.init
        getPageViewItem
        renderPage
        loop
        Frame.loop
        renderPagesUpto
        resize
```

## Describe the solution you'd like

If possible, a message that communicates more detail and outlines potential issues would be very helpful here.

**Comment by MurakamiShinyu** [comment]
@Enteleform Thank you for the report!

This error message shows there is a bug in vivliostyle.js.

```
 TypeError: Cannot read properties of null (reading 'funcs')
    at e.evalCall (file:///D:/DemoProject/node_modules/@vivliostyle/viewer/lib/js/vivliostyle-viewer.js:14:119825)  
```

The source code of this error location is:

https://github.com/vivliostyle/vivliostyle.js/blob/c0b5fca136c6ced50e0b637b0f89de95ef9aae0d/packages/core/src/vivliostyle/exprs.ts#L439-L446

The problem is the `scope` is null. This is obvious bug in the program.

Missing CSS Variables should not cause an error. The property value that has undefined variables have to be treated as `unset`. I tested a few such cases and did not find this problem.

I will investigate further and fix it as soon as possible.

**Comment by MurakamiShinyu** [comment]
@Enteleform could you give us a reproducible HTML+CSS example? Thanks!

**Comment by Enteleform** [comment]
@MurakamiShinyu

Sure thing!  
https://github.com/enteleform-codesandbox/0021--Vivliostyle--MissingVariables

The actual project had a whole page of missing CSS, but I just bisected it a few times and narrowed it down to a single `calc` expression, then simplified it to the most minimal cases:

```css
.Text{
  /** Build fails when variable is used in `calc`. **/
  margin: calc(var(--Undefined) / 2);

  /** Build fails just from presence of variable in `calc`. **/
  margin: calc(var(--Undefined));

  /** Build succeeds when variable is used directly. **/
  margin: var(--Undefined);

  /** Build succeeds when variable is used in certain properties. **/
  width: calc(var(--Undefined) / 2);
}

```

**Comment by MurakamiShinyu** [comment]
@Enteleform Thanks. I confirmed the error, that occurs when a shorthand property (e.g., `margin`) has undefined variable in calc expression.

**Comment by MurakamiShinyu** [comment]
Fixed in #1029 

Test:
- Before this fix: https://vivliostyle.github.io/viewer/v2.19.0/#src=https://github.com/enteleform-codesandbox/0021--Vivliostyle--MissingVariables/blob/main/index.html
- After this fix: https://vivliostyle-git-fix-issue1028-vivliostyle.vercel.app/#src=https://github.com/enteleform-codesandbox/0021--Vivliostyle--MissingVariables/blob/main/index.html
